src/Form/Form.js

```diff
@@ -3,27 +3,54 @@ import classes from "./Form.module.css";
 import * as firebase from "../lib/firebase";
 
 const Form = props => {
+  const { setdbItems, dbItems, token, onEnterToken } = props;
   const [name, setName] = useState("");
   const [frequency, setFrequency] = useState(0);
   const [date, setDate] = useState("");
 
   const handleSubmit = e => {
     e.preventDefault();
     let db = firebase.fb.firestore();
-    let data = {
-      name,
-      frequency,
-      date
-    };
-    db.collection(props.token)
-      .add(data)
-      .then(function() {
-        alert("Item added!");
-      })
-      .catch(function(error) {
-        alert("Something went wrong!");
-        console.error("Error writing document: ", error);
-      });
+    let newName = name
+      .toUpperCase()
+      .replace(/[^\w\s]|_/g, "")
+      .replace(/\s+/g, " ");
+    const existingItem = dbItems.find(element => {
+      let newElement = element.name
+        .toUpperCase()
+        .replace(/[^\w\s]|_/g, "")
+        .replace(/\s+/g, " ");
```

Comment:
I see we're doing this cleanup twice. This is fine as is, but consider pulling this out into a helper function that can be called in both places to avoid duplication.

Reply:
Also, is it worth cleaning up the `dbItems` outside of the find? What's your intention here? Does it need to be in the un-cleaned-up state?

Reply:
Our intention was to compare the cleaned up dbItem to the cleaned up user input, then return undefined or a match. For that, we only needed cleaned up dbItems in the find to compare with the user input since we're not using it for anything else. 

But I could definitely not be understanding how this works and totally be missing something.

Reply:
That makes complete sense. Thanks for clarifying.

Reply:
Maybe that could have been another good place for a comment?